internal/pkg/gateway/endpoint.go

```diff
@@ -90,7 +101,7 @@ func (ef *endpointFactory) newOrderer(address, mspid string, tlsRootCerts [][]by
 	return &orderer{
 		client:          connectOrderer(conn),
 		closeConnection: conn.Close,
-		endpointConfig:  &endpointConfig{address: address, mspid: mspid, tlsRootCerts: tlsRootCerts},
+		endpointConfig:  &endpointConfig{address: address, logAddress: logAddess, mspid: mspid, tlsRootCerts: tlsRootCerts},
```

Comment:
I think,  endpointConfig: `address: address` should be holding the connAddress right? (if it is overridden then `address` is not the actual endpoint address)

Reply:
No, the connAddress is only used to make the actual connection (on line 93 above).  The address in the endpointConfig structure must be the one that is returned by the discovery service because a future config update would need to match this address to determine whether the connection should be closed or not.